src/main/java/com/ss/training/lms/controller/LibrarianController.java

```diff
@@ -25,139 +18,116 @@
+    //     if (branches == null){
+    //         status = HttpStatus.NOT_FOUND;
+    //     }
+    //     return new ResponseEntity<List<LibraryBranch>>(branches , status);
+    // }
+
+    // @RequestMapping(path="/lms/librarian/books/{search}",
+    //                 produces = {
+    //                     MediaType.APPLICATION_XML_VALUE,
+    //                     MediaType.APPLICATION_JSON_VALUE
+    //                 })
+    // public ResponseEntity<List<Book>> getBooksWithSearch(@PathVariable String search)
+    // {
+    //     List<Book> books = null;;
+    //     HttpStatus status = HttpStatus.OK;
+
+    //     try {
+    //         books = librarianService.getBooksWithSearch(search);
+    //         if (books == null)
+    //         {
+    //             status = HttpStatus.NOT_FOUND;
+    //         }
+    //     } catch (SQLException e) {
+    //         e.printStackTrace();
+    //         return new ResponseEntity<List<Book>>(books , HttpStatus.INTERNAL_SERVER_ERROR);
+
+    //     }
+    //     return new ResponseEntity<List<Book>>(books , status);
+    // }
+
+    // @RequestMapping(path="lms/librarian/branches/{branch}/books/{book}/copies",
+    //                 produces = {
+    //                     MediaType.APPLICATION_XML_VALUE,
+    //                     MediaType.APPLICATION_JSON_VALUE
+    //                 })
+    // public ResponseEntity<BookCopies> getAnEntryOfBookCopies(@PathVariable int branch, @PathVariable int book)
+    // {
+    //     HttpStatus status = HttpStatus.OK;
+    //     BookCopies bookCopies = librarianService.getAnEntryOfBookCopies(branch, book);
+    //     if (bookCopies == null)
+    //     {
+    //         status = HttpStatus.NOT_FOUND;
+    //     }
+    //     return new ResponseEntity<BookCopies>(bookCopies , status);
+
+    // }
+
+    // @GetMapping(path="lms/librarian/branches/{branchId}/books")
+    // public ResponseEntity<List<Book>>getBooksAtABranch(@PathVariable int branchId)
+    // {
+    //     List<Book> books = null;
+    //     HttpStatus status = HttpStatus.OK;
+
+    //     try {
+    //         books = librarianService.getBooksAtABranch(branchId);
+    //         if (books == null)
+    //         {
+    //             status = HttpStatus.NOT_FOUND;
+    //         }
+    //     } catch (SQLException e) {
+    //         e.printStackTrace();
+    //         return new ResponseEntity<List<Book>>(books, HttpStatus.INTERNAL_SERVER_ERROR);
+
+    //     }
+    //     return new ResponseEntity<List<Book>>(books, status);
+
+    // }
 
     @PutMapping(value = "/lms/librarian/branches/{branchId}/books/{bookId}")
-    public ResponseEntity<BookCopies> updateCopies(@PathVariable int branchId,
+    public ResponseEntity<String> updateCopies(@PathVariable int branchId,
                                                @PathVariable int bookId,
                                                @RequestBody BookCopies bookCopies)
```

Comment:
Something I didn't think of before: What if the bookCopies object has a different branchId or bookId than those given as path parameters? Currently, the path parameters aren't actually used. We could just get rid of them and use a single path for this method, or we could change the request body parameter to just an integer (the new number of copies), or we could even add the number of copies as a new path parameter, though that seems somewhat counter-intuitive.